src/collectives/device/msccl_interpreter.h

```diff
@@ -178,8 +178,7 @@ namespace {
         }
         if (msccltran->has_dependence && tid == nthreads-1){
 	        if (needsFence) __threadfence();
-          uint64_t curFlag = COMPUTE_FLAG(workIndex, iter, step);
-          mscclFlags[bid].flag = curFlag;
+          mscclFlags[bid].flag = COMPUTE_FLAG(workIndex, iter, step);
```

Comment:
please cast this to (uint64_t) before storing it

Reply:
done!

Reply:
have you pushed it?